pygmt/datasets/load_remote_dataset.py

```diff
@@ -108,6 +108,25 @@ class GMTRemoteDataset(NamedTuple):
             "01m": Resolution(["gridline"], True),
         },
     ),
+    "earth_magnetic_anomaly": GMTRemoteDataset(
+        title="Earth magnetic anomaly",
+        name="magnetic_anomaly",
+        long_name="Earth magnetic anomaly",
+        units="nT",
+        extra_attributes={"horizontal_datum": "WGS84"},
```

Comment:
I'm not an expert in magnetic anomalies, but having a `horizontal_datum` seems strange. Is the magnetic anomaly referenced to the geoid somehow?

Also, not related to this PR, but I notice that `horizontal_datum` is also set as an attribute for the `earth_age` dataset at L96. Is seafloor crustal age also measured relative to a datum?

Reply:
Please see my comment https://github.com/GenericMappingTools/pygmt/pull/2200#discussion_r1038189103.

I think `horizontal_datum` means the coordinate system for longitude/latitude.

Reply:
AFAIK, the datum in GMT is mainly related to the ellipsoid and its positioning. As you can see from `gmt mapproject -Qd`, the positioning of the ellipsoid is given as the difference in three directions between the center of the ellipsoid and the center of the earth. As to `earth_age`, I think it does not have height information, `horizontal_datum` is enough. I don't konw much about magnetic anomalies.

Reply:
I'm going to leave `horizontal_datum` in; per my understanding (and from the comments above) it is the system that aligns coordinates with the actual locations on Earth. So I assume it still applies to the magnetic anomaly map?